src/automaton.ts

```diff
@@ -480,27 +502,22 @@ function applyCut(set: readonly Pos[]): readonly Pos[] {
   return found || set
 }
 
-function canMergeInner(a: State, b: State, mapping: readonly number[]) {
+function canMerge(a: State, b: State, mapping: readonly number[]) {
   for (let goto of a.goto) for (let other of b.goto) {
     if (goto.term == other.term && mapping[goto.target.id] != mapping[other.target.id]) return false
   }
   actions: for (let action of a.actions) {
-    let conflict = false
```

Comment:
This changes what this loop does—in the existing code, the action will be accepted (via `continue actions`) if any matching action is found. In the new code, the function will return false as soon as any conflicting action is found.

Given this distinction, I also don't think your assumption that the function will return the same for swapped `a`/`b` arguments is solid. It is probably possible to optimize that better than just calling it twice, though.

Reply:
ok reverted

Reply:
For what it's worth, this change didn't affect the generator output. Is it possible that this change wouldn't affect the overall algorithm?

Reply:
This will only make a difference for states with multiple actions for a token (GLR) that happen to be mergeable, so it is likely this change doesn't affect most grammars.